.github/workflows/actionlint.yaml

```diff
@@ -25,7 +25,7 @@ jobs:
     runs-on: ubuntu-24.04
     steps:
       - name: Checkout the code
-        uses: actions/checkout@v4
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
```

Comment:
nit:
```suggestion
        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
```

Reply:
Do you want that update for every hash? Did you use a different tool to do this in #11045 as it will take a long time to do manually!

Reply:
I started trying to modify the tool to get the output as suggested and then realised that although there are a lot of changes in this PR, it's only actually 5 actions, so doing find-replace and manual look ups wasn't too bad at all. Thanks for the improvement suggestion!